### PR TITLE
Come back home safely

### LINKED ISSUE #340: Come back home after run
We could finish run movements in a "start position". Then it should be easy to create a sharp run sequence as the inverse of the original sharp sequence. Then create a smooth path for this inverse run sequence to go back to the start cell.

### PR DESCRIPTION
**Pull request by Peque**
Closes #340, #357, #358.

### CODE REVIEW

File: src/target/move.c [hunk 1]
```diff
@@ -49,16 +49,16 @@ static void entered_next_cell(void)
  * This functions assumes the current speed is the target speed and takes into
  * account the configured linear deceleration.
  */
```
Comment (cua-cua) [inline src/target/move.c]: Include  @param[in] and @return comments

Comment (cua-cua) [inline src/target/move.c]: include @param, @return

File: src/solve.c [hunk 2]
```diff
@@ -189,8 +189,10 @@ static void execute_movement_sequence(char *sequence, uint8_t speed,
 			LOG_ERROR("Unable to process command [%d]!", movement);
 			return;
 		}
-		if (collision_detected())
+		if (collision_detected()) {
+			LOG_ERROR("Collision detected!");
```
[the remark this answers is not in the document]
Comment (Peque) [reply src/solve.c]: :woman_shrugging:

File: src/target/move.c [hunk 3]
```diff
@@ -286,6 +286,31 @@ void turn_back(uint8_t speed)
...
+/**
+ * @brief Turn back (180-degree turn) to a starting position.
+ *
+ * @param[in] speed The speed level at which to turn.
+ */
+void turn_to_start_position(uint8_t speed)
+{
+	float distance;
+
+	set_linear_acceleration(get_linear_acceleration() / 4.);
```
[the remark this answers is not in the document]
Comment (Peque) [reply src/target/move.c]: To try to adjust the front distance a bit more "smoothly" (avoiding slips by reducing the acceleration).

Maybe it is a bit easier to control this way and we may avoid some "step dances". We will see when we start analyzing the real-time logs! :blush:

File: src/target/move.c (hunk 1 above)
Comment (Peque) [reply src/target/move.c]: Reviews bellow the line you are reviewing! :stuck_out_tongue_winking_eye:

File: src/target/move.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (cua-cua) [reply src/target/move.c]: ok, :)

File: src/solve.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (cua-cua) [reply src/solve.c]: ok, we will see :)

File: src/target/move.c (hunk 1 above)
Comment (Peque) [reply src/target/move.c]: 300a8c4a10eacf8547df3cbbf3186a57ee088e05

File: src/solve.c [hunk 4]
```diff
@@ -187,6 +194,54 @@ void run(uint8_t speed)
...
+			break;
+		case 'S':
+			translation = 'B';
+			break;
+		default:
+			break;
+		}
+		run_back[length - i - 1] = translation;
+	}
+	run_back[length] = '\0';
```
[the remark this answers is not in the document]
Comment (Peque) [reply src/solve.c]: - 18ff7eb06731e04961ccd11021c84d81435b13f7
- edeec891e77193d6515b92cbf6617114b9c6e8d0

[the remark this answers is not in the document]
Comment (cua-cua) [reply src/solve.c]: :+1:

File: src/target/move.c (hunk 1 above)
Comment (cua-cua) [reply src/target/move.c]: ok

**Pull request merged**